### ISSUE TITLE
support dimension (time) for WMTS 1.0.0

### ISSUE THREAD

**Issue by murdav** [geopython/OWSLib#368]
Dear All,

Following http://portal.opengeospatial.org/files/?artifact_id=35326 there is the possibility to have the dimension, page 22 Table 9 — Parts of Dimension data structure.

e.g.
```
<Style isDefault="true">
  <ows:Title xml:lang="en">default</ows:Title>
  <ows:Identifier>default</ows:Identifier>
</Style>
<Format>image/png</Format>
<Dimension>
  <ows:Identifier>time</ows:Identifier>
  <ows:UOM>ISO8601</ows:UOM>
  <Default>2017-05-16</Default>
  <Current>false</Current>
  <Value>2015-07-30/2017-05-16/P1D</Value>
</Dimension>

```
A getcapabilities example could be:

https://gibs-b.earthdata.nasa.gov/wmts/epsg4326/best/wmts.cgi?Service=WMTS&Request=getcapabilities&Version=1.0.0

If there is no program to support it, I can have a look at it. 

@tomkralidis the issue is that the WMS-T has a different structure (page 53, 06-042_OpenGIS_Web_Map_Service_WMS_Implementation_Specification.pdf),  could you give me a bit of help, maybe some points of start?

Thanks,

D

**Comment by tomkralidis** [comment]
@murdav looking at http://schemas.opengis.net/wmts/1.0/wmtsGetCapabilities_response.xsd, `wmts:Dimension` occurs 0..n times in a given `Layer` element. 

![wmts-layer-dimension](https://cloud.githubusercontent.com/assets/910430/26151901/c90fc700-3ad2-11e7-8f49-25ea85bdebc1.jpg)

 I would create a `dimensions` property which is a list of each dimension with the appropriate properties, starting at around https://github.com/geopython/OWSLib/blob/master/owslib/wmts.py#L773.  The `wmts:Value` property being a list of values.

**Comment by murdav** [comment]
Thanks for this! 
Further question, I didn't find how I can run OWSLib tests before pulling the request from my fork to master. I got some minor errors (like field title has changed the value) that I can easily update. 
Is it possible to run the tests locally, if so, how?

D

**Comment by tomkralidis** [comment]
@murdav you can run the tests with py.test (see how Travis does it in https://github.com/geopython/OWSLib/blob/master/.travis.yml#L17).  Note you can also run tests on single doctest files.

**Comment by acerioni** [comment]
+1 for this feature request!

**Comment by lubojr** [comment]
Feature is implemented in #904 and for now can be used from a fork https://github.com/lubojr/OWSLib/tree/368-wmts-dimensions-support if anyone needs it

**Comment by tomkralidis** [comment]
Thanks @lubojr.  Merged.
